In [5]:
import pandas as pd
import numpy as np
import math

# Read in the data
df = pd.read_csv('./Data/HPO_input/Results/GPT4_top_prediction_RE.csv')
# split columns into multiple columns
df['dataset'] = df['file_path'].str.split('/', expand=True)[5]

In [6]:
# format float to 2 decimal places
w = df.shape[0]
x = df[['predict_correct_in_top_5']].values.sum()
y = df[['predict_made_in_top_10']].values.sum()
print('{y} out of {w} predictions made in top 5 ({z}%)'.format(y=y, w=w, z=np.round(y/w,4)*100))
print('{x} out of {y} predictions correct in top 5 ({z}%)'.format(x=x, y=y, z=np.round(x/y,4)*100))

x = df[['predict_correct_in_top_10']].values.sum()
y = df[['predict_made_in_top_10']].values.sum()
print('{y} out of {w} predictions made in top 10 ({z}%)'.format(y=y, w=w, z=np.round(y/w,4)*100))
print('{x} out of {y} predictions correct in top 10 ({z}%)'.format(x=x, y=y, z=np.round(x/y,4)*100))

x = df[['predict_correct_in_top_50']].values.sum()
y = df[['predict_made_in_top_50']].values.sum()
print('{y} out of {w} predictions made in top 50 ({z}%)'.format(y=y, w=w, z=np.round(y/w,4)*100))
print('{x} out of {y} predictions correct in top 50 ({z}%)'.format(x=x, y=y, z=np.round(x/y,4)*100))


219 out of 276 predictions made in top 5 (79.35%)
4 out of 219 predictions correct in top 5 (1.83%)
219 out of 276 predictions made in top 10 (79.35%)
16 out of 219 predictions correct in top 10 (7.31%)
204 out of 276 predictions made in top 50 (73.91%)
21 out of 204 predictions correct in top 50 (10.290000000000001%)


In [41]:
# subset df by dataset
dataset_list = df['dataset'].unique()
for dataset in dataset_list:
    print(dataset)
    df_subset = df[df['dataset']==dataset]
    w = df_subset.shape[0]
    x = df_subset[['predict_correct_in_top_5']].values.sum()
    y = df_subset[['predict_made_in_top_5']].values.sum()
    print('{y} out of {w} predictions made in top 5 ({z}%)'.format(y=y, w=w, z=np.round(y/w,4)*100))
    print('{x} out of {y} predictions correct in top 5 ({z}%)'.format(x=x, y=y, z=np.round(x/y,4)*100))

    x = df_subset[['predict_correct_in_top_10']].values.sum()
    y = df_subset[['predict_made_in_top_10']].values.sum()
    print('{y} out of {w} predictions made in top 10 ({z}%)'.format(y=y, w=w, z=np.round(y/w,4)*100))
    print('{x} out of {y} predictions correct in top 10 ({z}%)'.format(x=x, y=y, z=np.round(x/y,4)*100))

    x = df_subset[['predict_correct_in_top_50']].values.sum()
    y = df_subset[['predict_made_in_top_50']].values.sum()
    print('{y} out of {w} predictions made in top 50 ({z}%)'.format(y=y, w=w, z=np.round(y/w,4)*100))
    print('{x} out of {y} predictions correct in top 50 ({z}%)'.format(x=x, y=y, z=np.round(x/y,4)*100))
    print('')


ColumbiaU
27 out of 27 predictions made in top 5 (100.0%)
6 out of 27 predictions correct in top 5 (22.220000000000002%)
27 out of 27 predictions made in top 10 (100.0%)
7 out of 27 predictions correct in top 10 (25.929999999999996%)
22 out of 27 predictions made in top 50 (81.47999999999999%)
4 out of 22 predictions correct in top 50 (18.18%)

CSH
70 out of 72 predictions made in top 5 (97.22%)
12 out of 70 predictions correct in top 5 (17.14%)
63 out of 72 predictions made in top 10 (87.5%)
13 out of 63 predictions correct in top 10 (20.630000000000003%)
49 out of 72 predictions made in top 50 (68.06%)
10 out of 49 predictions correct in top 50 (20.41%)

AJHG
74 out of 78 predictions made in top 5 (94.87%)
0 out of 74 predictions correct in top 5 (0.0%)
71 out of 78 predictions made in top 10 (91.03%)
0 out of 71 predictions correct in top 10 (0.0%)
51 out of 78 predictions made in top 50 (65.38000000000001%)
1 out of 51 predictions correct in top 50 (1.96%)

DGD
82 out of 85 predict

In [51]:
def probability_of_picking_specific_card(total_cards, specific_cards, draws):
    probability_not_picking = 1

    for i in range(draws):
        non_specific_cards = total_cards - specific_cards - i
        remaining_cards = total_cards - i
        probability_not_picking *= non_specific_cards / remaining_cards

    probability_picking = 1 - probability_not_picking
    return probability_picking
# calculate the probability of select the correct gene in top 5

total_cards = 30000
specific_cards = 1
# what is the probability of selecting the correct gene in top 5
draws = 5
probability = probability_of_picking_specific_card(total_cards, specific_cards, draws)
print('Probability of selecting the correct gene in top 5: {}%'.format(probability*100))
draws = 10
probability = probability_of_picking_specific_card(total_cards, specific_cards, draws)
print('Probability of selecting the correct gene in top 10: {}%'.format(probability*100))
draws = 50
probability = probability_of_picking_specific_card(total_cards, specific_cards, draws)
print('Probability of selecting the correct gene in top 50: {}%'.format(probability*100))

Probability of selecting the correct gene in top 5: 0.01666666666666483%
Probability of selecting the correct gene in top 10: 0.033333333333340764%
Probability of selecting the correct gene in top 50: 0.16666666666667052%


In [52]:
# This table is generated using ChatGPT-3

| Dataset   | Top 5          |                | Top 10         |                | Top 50          |                |
|-----------|----------------|-------------- |----------------|-------------- |-----------------|-------------- |
|           | Predictions (%)| Correct (%)   | Predictions (%)| Correct (%)   | Predictions (%)| Correct (%)   |
| ColumbiaU | 27 (100.00%)   | 6 (22.22%)    | 27 (100.00%)   | 7 (25.93%)    | 22 (81.48%)     | 4 (18.18%)    |
| CSH       | 70 (97.22%)    | 12 (17.14%)   | 63 (87.50%)    | 13 (20.63%)   | 49 (68.06%)     | 10 (20.41%)   |
| AJHG      | 74 (94.87%)    | 0 (0.00%)     | 71 (91.03%)    | 0 (0.00%)     | 51 (65.38%)     | 1 (1.96%)     |
| DGD       | 82 (96.47%)    | 10 (12.20%)   | 81 (95.29%)    | 14 (17.28%)   | 62 (72.94%)     | 6 (9.68%)     |
| TAF1      | 8 (57.14%)     | 0 (0.00%)     | 11 (78.57%)    | 0 (0.00%)     | 11 (78.57%)     | 0 (0.00%)     |
